src/utils/analysis-utils.ts

```diff
@@ -648,3 +671,59 @@ export class BaselineLayerResult {
       : `${this.getHazardLayer().title} (${this.statistic})`;
   }
 }
+
+export class PolygonAnalysisResult {
+  key: number = Date.now();
+  featureCollection: FeatureCollection;
+  tableData: TableRow[];
+  tableColumns: string[];
+
+  statistic: 'area' | 'percentage';
+  threshold?: ThresholdDefinition;
+  legend: LegendDefinition;
+  legendText: string;
+  hazardLayerId: WMSLayerProps['id'];
+  adminLevel: AdminLevelType;
+
+  constructor(
+    tableData: TableRow[],
+    tableColumns: string[],
+    featureCollection: FeatureCollection,
+    hazardLayer: WMSLayerProps,
+    adminLevel: AdminLevelType,
+    statistic: 'area' | 'percentage',
+    threshold?: ThresholdDefinition,
+  ) {
+    this.featureCollection = featureCollection;
+    this.tableData = tableData;
+    this.tableColumns = tableColumns;
+    this.statistic = statistic;
+    this.threshold = threshold;
+    this.adminLevel = adminLevel;
+
+    // color breaks from https://colorbrewer2.org/#type=sequential&scheme=Reds&n=5
+    this.legend = [
+      { label: '20%', value: 0.2, color: '#fee5d9' },
+      { label: '40%', value: 0.4, color: '#fcae91' },
+      { label: '60%', value: 0.6, color: '#fb6a4a' },
+      { label: '80%', value: 0.8, color: '#de2d26' },
+      { label: '100%', value: 1, color: '#a50f15' },
```

Comment:
consider commenting what all these colors are, or use `hsl()` since that's easier for humans to read. Or even try using MUI colors (MUI library and theme comes with a lot of pre-defined colors)

Reply:
I'm hesitant to use hsl because as far as I can tell, hsl isn't currently being used in the code base and I'm trying to keep with the established practice of using hex.  I'd love to learn more about the hsl function.  Honestly, rgb is the easiest for me to read, but that might be my own personal preference.  I think I can add some comments describing the hsl values, so at least people can read them better.  I'm open to standardizing and potentially refactoring how we do colors, but I think that when we do so, it should happen everywhere and that should be a separate PR.

Reply:
Yup thats fair. See if you can use MUI colors at least?
e.g:
```ts
import { red } from '@material-ui/core/colors';
//...
color: red['50'],
// ...
```

Reply:
addressed with https://github.com/WFP-VAM/prism-frontend/pull/431/commits/937cf00fd8ad7d52a573708887ffb67c71fb701e.  

Here are all the base colors on mui:
<img width="274" alt="Screen Shot 2022-04-02 at 11 46 21 PM" src="https://user-images.githubusercontent.com/4313463/161410457-8eb8aecc-3edd-4a43-8170-e496c2c55471.png">

I went through all the red ones:
<img width="305" alt="Screen Shot 2022-04-02 at 11 46 51 PM" src="https://user-images.githubusercontent.com/4313463/161410467-9c86f9f5-cf21-455f-be7d-0f5a7794ca28.png">

Although there are matches for some of the red ones, the color scale doesn't go as dark as the one by colorbrewer2.org

I googled each color currently being used, trying to find any common term or description for it.  I then added a name (or short description) to the comment for each color.

Resolve if this fixes the issue.  Otherwise, let's keep talking.

Thank you.

Reply:
Great! Thanks, sorry if it was a little tiresome getting this resolved, all good now